app/models/document.rb

```diff
@@ -277,8 +276,7 @@ def audit_date_comment
 
   def reset_replacement_document_validation_request_update_counter!
     return unless validated? || archived?
-
-    return unless (request = ReplacementDocumentValidationRequest.find_by(new_document_id: id))
+    return unless (request = owner)
 
     request.reset_update_counter!
```

Comment:
A little concerned that the assignment here is too easily mistaken for a comparison (since assignment in a condition is unusual).

Maybe more like:

```ruby
    return if owner.nil?

    owner.reset_update_counter!
```